lndr-cli/src/Lndr/CLI/Actions.hs

```diff
@@ -312,12 +313,30 @@ submitCredit url secretKey unsignedCredit@(CreditRecord creditor debtor _ _ _ _
                    then HTTP.parseRequest $ url ++ "/lend"
                    else HTTP.parseRequest $ url ++ "/borrow"
     let signedCredit = signCredit secretKey (unsignedCredit { nonce = nonce })
-    let req = HTTP.setRequestBodyJSON signedCredit $
+        req = HTTP.setRequestBodyJSON signedCredit $
                 HTTP.setRequestMethod "POST" initReq
     resp <- HTTP.httpNoBody req
     return $ HTTP.getResponseStatusCode resp
 
 
+submitMultiSettlement :: String -> Text -> [CreditRecord] -> IO Int
+submitMultiSettlement url secretKey transactions = do
+    startNonce <- getNonce url (debtor $ head transactions) (creditor $ head transactions)
+    initReq <- HTTP.parseRequest $ url ++ "/multi_settlement"
+    let signedTransactions = signMultiSettlement secretKey startNonce transactions []
+        req = HTTP.setRequestBodyJSON signedTransactions $
+                HTTP.setRequestMethod "POST" initReq
+    resp <- HTTP.httpNoBody req
+    return $ HTTP.getResponseStatusCode resp
+
+
+signMultiSettlement :: Text -> Integer -> [CreditRecord] -> [CreditRecord] -> [CreditRecord]
```

Comment:
- Do not make a top level function to do this, put it in a `where` block in `submitMultiSettlement`.
- Do not write a general recursive function to do this. This can be done in a much more readable way with a fold--just use an accumulator that is a pair of nonce and signed txs.
- If you have to call `reverse` at the end of a function, there's probably a cleaner way to do it, or perhaps a better data structure to use. Look into the differences between `foldl` and `foldr` in this regard.